### DISCUSSION TITLE
[Core Protocol] op-proposer: What is the purpose of designing "withdrawal_storage_root"?

### DISCUSSION THREAD

**Discussion by joohhnnn** [ethereum-optimism/developers#16] [Core Protocol] [answered]
### Issue Description

https://github.com/ethereum-optimism/optimism/blob/develop/specs/proposals.md#l2-output-commitment-construction
In this section, there's a term "withdrawal_storage_root". I found it in the codebase, but it doesn't seem to be used anywhere. Is it going to be used in the future, or does it have any specific purpose?
<img width="644" alt="image" src="https://github.com/ethereum-optimism/developers/assets/68833933/0b6cd824-1481-4230-b456-6ddbb2381436">

**Comment by sbvegan** [comment]
Hey @joohhnnn, I'm not sure about this particular detail. I'll work on finding someone with more domain specific knowledge to get the answer for you.

**Comment by ajsutton** [comment, marked as answer]
The withdrawal storage root is just "hoisting" the root of the withdrawal contract up to be a top level part of the output root hash.  That way merkle proofs of withdrawals are much smaller since they don't need to provide as many levels of data. So it would be used as part of the L1 contracts to prove withdrawals, but it wouldn't have that name there since it's just working with a hash and merkle proof.

**Reply by sbvegan** [reply]
@joohhnnn does this answer your question?

**Reply by joohhnnn** [reply]
thanks! It's very clear now 👍
